learning-basics/configure-datastructures-at-runtime/src/main/java/ClientDynamicConfig.java

```diff
@@ -0,0 +1,46 @@
+import com.hazelcast.client.HazelcastClient;
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.IMap;
+import com.hazelcast.logging.ILogger;
+
+import static com.hazelcast.config.MapStoreConfig.InitialLoadMode.EAGER;
+
+public class ClientDynamicConfig {
+
+    /*
+     * Starts a Hazelcast member and client without any explicit data structure configuration, then adds
+     * dynamic data structure configuration from a client
+     */
+    public static void main(String[] args) {
+        // Start a member with no explicit configuration
+        HazelcastInstance instance = Hazelcast.newHazelcastInstance();
+        ILogger logger = instance.getLoggingService().getLogger(ClientDynamicConfig.class);
+
+        // Start a client -- we will be using the map to access & configure data structures
+        HazelcastInstance client = HazelcastClient.newHazelcastClient();
+        // Obtain a map
+        IMap<String, String> defaultMap = client.getMap("defaultMap");
+        // no data have been added yet, get("1") will return null
```

Comment:
`// no data has been added yet, so get("1") will return null`

Reply:
You may be interested in the "is data singular or plural" debate 😉  : https://www.theguardian.com/news/datablog/2010/jul/16/data-plural-singular